portfolio/src/main/webapp/request.js

```diff
@@ -25,4 +25,88 @@ class ScheduleRequest {
   }
 }
 
-// TODO(tomasalvarez): write function to fetch results and display results
+/**
+ * Gets all of the scheduling information from the UI and returns a
+ * ScheduleRequest with all of the data.
+ */
+function createRequestFromUiInformation() {
+  const startTime = document.getElementById('working-hour-start').value;
+  const endTime = document.getElementById('working-hour-end').value;
+  const events = collectAllEvents();
+  const tasks = collectAllTasks();
+  const algorithmType = document.getElementById('algorithm-type').value;
+  const scheduleRequest = new ScheduleRequest(
+      events, tasks, getTimeObject(startTime), getTimeObject(endTime),
+      algorithmType);
+  return scheduleRequest;
+}
+
+/**
+ * Gets called when the user hits 'Start Scheduling'.
+ */
+function startScheduling() {
+  const scheduleRequest = createRequestFromUiInformation();
+  // Create the request to send to the server using the data we collected from
+  // the web form.
+  collectAllScheduledTasks(scheduleRequest).then((scheduledTaskArray) => {
```

Comment:
Do you think if you could store the scheduledTaskArray into its variable once it is received from the server, and return this array somehow? I think having a getter method that returns the scheduledTaskArray will be helpful for updating and exporting to Calendar and Tasks later.

Reply:
I broke the up functionality and added a new local variable scheduledTasks. One function will now fetch from the servlet and set the local variable while another function simply returns this local variable.